routes/web.php

```diff
@@ -20,8 +20,27 @@
 Route::get('', 'HomeController@index')->name('index');
 Route::get('about', 'HomeController@aboutPage')->name('about');
 Route::post('kmlpolygonsearch', 'GazetteerController@searchFromKmlPolygon')->name('searchFromKmlPolygon'); //search from file
-Route::get('places/{id?}', 'GazetteerController@search')->name('places'); //shows places with optional id, if no id is given it uses all results before applying filters
-Route::get('search', 'GazetteerController@search')->name('search')->middleware('checkmaxpaging', 'cors');
+
+// Redirect from search to places
+Route::get('search', function (Request $request) {
+    $queryParameters = $request->query();  // Get all query parameters
+
+    $redirectURL = '/places';
+
+    // if ($id = $request->input('id')) {
+    //     $redirectURL .= '/' . $id;  //TODO  comfirm this
+    // }
+
+    // Build query string and append it to redirect URL
+    if (!empty($queryParameters)) {
+        $redirectURL .= '?' . http_build_query($queryParameters);
+    }
+
+    return redirect($redirectURL);
+});
+Route::get('places', 'GazetteerController@search')->name('places')->middleware('checkmaxpaging', 'cors'); 
+Route::get('places/{id?}', 'GazetteerController@search')->name('places'); 

```

Comment:
This route isn't working properly currently.

Reply:
It works well now. Please create a redirect from `places?id={place_id}` to `places/{place_id}`

Reply:
Can we create a redirect from search?id={id} to places/{place_id} instead? If we define redirects both from search to places and from places?id={place_id} to places/{place_id} it will likely cause a loop"

Reply:
Ideally should redirect from `places?id={place_id` because it is a valid route currently, and we probably don't want user to use the URL in this form.

Have a try to see if it works. Check the get parameter id from the request and redirect it to `places/{place_id}` if it presents.

Reply:
have tried : Route::get('places/{path?}', function (Request $request ,  $path = null) {
    $params = $request->all();
    if(isset($params['id'])){
        return redirect()->to("places/{$params['id']}");
    }
    return redirect()->to('/places/' . $path . '?' . $request->getQueryString());
})->middleware('checkmaxpaging', 'cors');    .  it can redirect from places?id={id} to places/{id}, but it will cause a loop by redirecting places to places.

Reply:
You need to solve this by removing the `id` query string once it's been rediected.

Reply:
Or separate it to two routes one `places` and another `places/{id}`

Reply:
Updated